File: exercise/java/day20/src/main/java/reindeer/model/ReindeerResult.java
package reindeer.model;

import java.util.UUID;

public class ReindeerResult {
    private final UUID id;
    private final String name;
    private final ReindeerColor color;

    private ReindeerResult(UUID id, String name, ReindeerColor color) {
        this.id = id;
        this.name = name;
        this.color = color;
    }

    public static ReindeerResult from(Reindeer reindeer) {
        return new ReindeerResult(reindeer.id(), reindeer.name(), reindeer.color());
    }

    public UUID getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public ReindeerColor getColor() {
        return color;
    }
}

File: exercise/C#/day20/Reindeer.Web/Service/ReindeerResult.cs
namespace Reindeer.Web.Service
{
    public class ReindeerResult
    {
        public Guid Id { get; }
        public string Name { get; }
        public ReindeerColor Color { get; }

        private ReindeerResult(Guid id, string name, ReindeerColor color)
        {
            Id = id;
            Name = name;
            Color = color;
        }

        public static ReindeerResult From(Reindeer reindeer) => new(reindeer.Id, reindeer.Name, reindeer.Color);
    }
}